lib/api/v3/work_packages/update_end_point.rb

```diff
@@ -3,6 +3,9 @@ module V3
     module WorkPackages
       class UpdateEndPoint < API::V3::Utilities::Endpoints::Update
         def present_success(current_user, call)
+          work_package = WorkPackage.find(call.result['id'])
```

Comment:
This seems like the best place to add the hook for catching the `patch api/v3/work_packages/:id` call.
Alternatively it could be put on the model so the logic also applies to other code paths? But I'm not super happy about after save callbacks.

Reply:
Hey @Pectojin, that place would theoretically work. However, it is not optimal. Let me explain an important detail. When a user updates a work package attribute a journal entry will be created and saved to the DB that logs the changes. Now imagine you change 5 attributes one after another, you don't want to send notification for each changed attribute. What we do is that we aggregate all changes and send only one notification per user after a defined period of time.

An exception is a work package comment. They always get sent immediately to keep a conversation flowing. 

So I think that it would be a good idea to apply the same logic to this feature for this feature here. So only when that waiting time has passed we send out the notification detailing which work package are now unblocked.

So you might want to consider subscribing to **OpenProject::Events::AGGREGATED_WORK_PACKAGE_JOURNAL_READY** and then checking the aggregated journal for status changes.

Reply:
The call has been moved in https://github.com/Pectojin/openproject/pull/1/commits/45eab16e37e86c6107adf1e54c8351a0211f5f00

There's quite a few more cases to consider because journals can contain many types of changes, but I think I considered the relevant cases.